added script to automatically perform CH4 -> CH3I alchemical simulati…

Lumber1432 opened the pull request:
…on with BSS

Added script to automate alchemical simulations between small non-flexible molecules, with default to methane -> iodomethane.
Script uses BioSimSpace (OpenBioSim), GAFF force-field and GROMACS to run the simulation.

Molecules are input as SMILES and duration of equilibration/production phases is input in a dictionary with values in picoseconds; script can be modified to change these values and it will flag calculation if it was too short to produce valid results. 

Attempted a calculation and estimated dG at 2.5 kcal/mol (expected 3.09 kcal/mol).

I excluded openfe because it refused to run on GPU and when it finally reached end of calculation it failed to estimate in vacuum. Documentation is not very thorough on relative free energy of solvation and the script already required some tweaking to render it possible.

Also added a small README to explain the script.

Lumber1432 committed:
added script to automatically perform CH4 -> CH3I alchemical simulation with BSS

Lumber1432 committed:
separated simulations in BSS and openfe

Lumber1432 commented:
I've added a new script for absolute hydration free energies (AHFE) and relative hydration free energies (RHFE) in `openfe`. We might need to use `openfe` as I couldn't find a way to perform AHFE in BSS and BSS does not support `espaloma` whereas [`openfe` does](https://docs.openfree.energy/en/v1.4.0/reference/api/openmm_protocol_settings.html). 

I've tried running some simulations with mixed results.

The AHFE simulation did not finish successfully; I believe this might be because I did not let it run for sufficient time (10ps for equilibration, 10ps for production), but I currently do not have a suitable machine to run this on, and `openfe` still refuses to acknowledge my GPU. I've been trying to get a machine from our local provider with little luck.

The RHFE simulation concluded successfully multiple times, yielding a ddG of $-1.6\pm 0.3$ kcal/mol, but this is quite far from the estimate of FreeSolv. More importantly, the problem is that all _in vacuo_ transformations always return a result of $0.0\pm 0.0$ kcal/mol, which I find suspicious. I was wondering if maybe I overlooked something? 

Thanks and see you soon, 
Giulia

andrrizzi reviewed (approved):
Looks awesome! Thank you!

I'd just update the `README.md` and then I think we're ready to go!

andrrizzi commented:
> I did not let it run for sufficient time (10ps for equilibration, 10ps for production)

This might indeed be quite short. I've just checked and the default parameters should be correct. We might be able to run a bit less in solvation, but let's try to do the whole 10ns if it's not too expensive.

> openfe still refuses to acknowledge my GPU

I'd double check that OpenMM can find your GPU. It might be easier to solve the problem. Could it be that the installed version is not compatible with the CUDA driver on your system?

> The RHFE simulation concluded successfully multiple times, yielding a ddG of −1.6±0.3 kcal/mol, but this is quite far from the estimate of FreeSolv. More importantly, the problem is that all in vacuo transformations always return a result of 0.0±0.0 kcal/mol, which I find suspicious.

Let's put aside the RHFE for now, but this is useful info. As a note for later (in case we need to revisit this, feel free to ignore this for now) BioSimSpace might use reaction field rather than PME, which usually can lead to quite different results.

You can in principle define a thermodynamic cycle that doesn't require the vacuum stage if you decouple the charges rather than annihilating them. In this case, I would expect the vacuum to return 0.0. If there's no bug, this might be the explanation, in which case we simply don't run vacuum.

Lumber1432 committed:
changes to README and some modifications to scripts

Lumber1432 commented:
I fixed the README.md to reflect changes, and modified the scripts so that simulation time is accessible, but the durations are now the default ones. I've tried installing different drivers/openfe versions but it really doesn't want to work. I am on an oldish laptop so I believe it might be a mix of hardware and software compatibility problems. If everything is okay with this commit, I think it's ready to be merged.
